clients/edgex-foundry/device_client.go

```diff
@@ -198,7 +202,12 @@ func (efc *EdgexDeviceClient) GetPropertyState(ctx context.Context, propertyName
 
 // getPropertyState returns different error messages according to the status code
 func getPropertyState(getURL string) (*resty.Response, error) {
-	resp, err := resty.New().R().Get(getURL)
+	cookieJar, _ := cookiejar.New(&cookiejar.Options{PublicSuffixList: publicsuffix.List})
+	// TODO: no need to instantiate a client for every request
+	resp, err := resty.NewWithClient(&http.Client{
+		Jar:     cookieJar,
```

Comment:
This client uses a cookie, but I don't see any other use of that cookie. Can you explain why we need it?

Reply:
The is the way which resty library's New() function create the http client. 
func New() *Client {
	cookieJar, _ := cookiejar.New(&cookiejar.Options{PublicSuffixList: publicsuffix.List})
	return createClient(&http.Client{
		Jar: cookieJar,
	})
}
I just set timeout for the client on this basis.